### PR TITLE
Add possibility to list Default Reviewers

### PR DESCRIPTION
**Pull request by j0nathan33**

### CODE REVIEW

**Comment by cdancy** [comment]
@j0nathan33 rebase off of master as I've made many changes, especially surrounding live tests which as I've noted before can now be run successfully OOTB. Take a look at the `@BeforeClass/init` and `@AfterClass/fin` methods to see how you can setup live tests.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        List<Condition> conditionList = api().listConditions(projectKey, repoKey);
+        assertThat(conditionList).isEmpty();
+    }
+
+    @AfterClass
+    public void fin() {
+        boolean success = api.repositoryApi().delete(projectKey, repoKey);
+        assertThat(success).isTrue();
+        success = api.projectApi().delete(projectKey);
+        assertThat(success).isTrue();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: Once you rebase off of master this will no longer be necessary as we now have a standard for cleaning up.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Repository repository = api.repositoryApi().create(projectKey, createRepository);
+        assertThat(repository).isNotNull();
+        assertThat(repository.errors().isEmpty()).isTrue();
+        assertThat(repoKey.equalsIgnoreCase(repository.name())).isTrue();
+    }
+
+    @Test
+    public void testListDefaultReviewersOnEmptyRepo() {
+        List<Condition> conditionList = api().listConditions(projectKey, repoKey);
+        assertThat(conditionList).isEmpty();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: Test for when it's not empty?

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        CreateProject createProject = CreateProject.create(projectKey, null, null, null);
+        Project project = api.projectApi().create(createProject);
+        assertThat(project).isNotNull();
+        assertThat(project.errors().isEmpty()).isTrue();
+        assertThat(project.key().equalsIgnoreCase(projectKey)).isTrue();
+        CreateRepository createRepository = CreateRepository.create(repoKey, true);
+        Repository repository = api.repositoryApi().create(projectKey, createRepository);
+        assertThat(repository).isNotNull();
+        assertThat(repository.errors().isEmpty()).isTrue();
+        assertThat(repoKey.equalsIgnoreCase(repository.name())).isTrue();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: Once you rebase off of master we do something similar to this behind the scenes thus making this unnecessary.

**Comment by j0nathan33** [comment]
Right now, I don't run IntegrationTest because It don't working :(. I will try to debug tomorrow

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java (hunk 2 above)
Comment (j0nathan33) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: It's under another PR, because I need to create some DefaultReviewer and when create new repo, the list is empty.

**Comment by cdancy** [comment]
@j0nathan33 `integTest` should work OOTB now assuming you have the correct gradle properties passed or set. I'd be interested to know what your failures are if you can share.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    private GeneratedTestContents generatedTestContents;
+
+
+    @BeforeClass
+    public void init() {
+        generatedTestContents = initGeneratedTestContents();
+    }
+
+    @Test
+    public void testListDefaultReviewersOnEmptyRepo() {
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: Can we come up with a better name here as we are only listing conditions and not reviewers? Is there anyway to add conditions so that we can check whether this actually works?

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        List<Condition> conditionList = api().listConditions(generatedTestContents.project.key(), generatedTestContents.repository.slug());
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: Also ... as your going to be building upon this for subsequent PR's set the `projectKey` and `repoKey` within the `init` method for all downstream tests to use. While this way works it will become very verbose as we write more tests. Take a look a the other live tests to see what I've done.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiMockTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by jdoire on 24/05/2017.
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiMockTest.java]: Lets not inject our own user information here and instead follow the pattern used by other live tests.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java (hunk 4 above)
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiLiveTest.java]: I stand corrected again. It would seem you do need to configure the default reviewers AOT prior to executing this otherwise it turns up empty. Hmmm ... not sure how we want to handle this then. You do have another PR to create default reviewers. I can merge this but I'd like you to add another test there which invokes this list again to ensure it is non-empty after the reviewer is added.

File: src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiMockTest.java (hunk 6 above)
Comment (j0nathan33) [inline src/test/java/com/cdancy/bitbucket/rest/features/DefaultReviewersApiMockTest.java]: Oops, I forgot to remove it
